near-sdk/tests/test-contracts/lazy/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::{near, store::LazyOption, PanicOnDefault};
+
+#[derive(BorshSerialize, BorshDeserialize, Ord, PartialOrd, Eq, PartialEq, Clone)]
+#[borsh(crate = "near_sdk::borsh")]
+pub struct Insertable {
+    pub index: u32,
+    pub data: String,
+    pub is_valid: bool,
+}
+
+#[near(contract_state)]
+#[derive(PanicOnDefault)]
+pub struct LazyContract {
+    pub lazy_opt: LazyOption<Insertable>,
+}
+
+#[near]
+impl LazyContract {
+    #[init]
+    pub fn new() -> Self {
+        let lazy_opt = LazyOption::new(b"a", None);
+        Self { lazy_opt }
+    }
+
+    fn insertable(&self) -> Insertable {
+        Insertable { index: 0, data: "scatter cinnamon wheel useless please rough situate iron eager noise try evolve runway neglect onion".to_string(), is_valid: true }
+    }
+
+    /// This should only write to the underlying storage once.
+    #[payable]
+    pub fn flush(&mut self, iterations: usize) {
+        let insertable = self.insertable();
+        self.lazy_opt.set(Some(insertable));
+
+        for _ in 0..=iterations {
+            self.lazy_opt.flush();
+        }
+    }
+
+    /// This should write on each iteration.
+    #[payable]
+    pub fn insert_flush(&mut self, iterations: u32) {
+        let mut insertable = self.insertable();
+        for idx in 0..=iterations {
+            insertable.index = idx as u32;
+            self.lazy_opt.set(Some(insertable.clone()));
+            self.lazy_opt.flush();
+        }
+    }
+
+    /// This should write and delete on each iteration.
+    #[payable]
```

Comment:
I know we use the same for store contracts but what's the motivation to make these functions `payable` in tests?
I think it's because payable inflicts extra cost, so we calculate on worse case scenario, but just to be on the same page